CMakeLists.txt

```diff
@@ -104,6 +110,11 @@ add_rust_feature(
     RUST_FEATURE "iceoryx2/logger_tracing"
 )
 
+if(TREAT_WARNING_AS_ERROR)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wextra -Wpedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra -Wpedantic")
+endif()
```

Comment:
We need to handle this closer to iceoryx classic and enable those warnings by default. Not for this PR, though.